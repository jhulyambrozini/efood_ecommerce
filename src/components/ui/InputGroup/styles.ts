import styled from 'styled-components'
import { colors } from '../../../styles'

type InputGroupProps = {
  maxWidth?: string
}

export const InputGroupContainer = styled.div<InputGroupProps>`
  display: grid;
  margin-bottom: 0.5rem;

  label {
    font-size: 0.875rem;
    padding-bottom: 0.5rem;
  }

  input {
    padding: 0.5rem;
    color: #4b4b4b;
    font-size: 0.875rem;
    border: none;
    background-color: ${colors.primaryLight};
    border: 2px solid ${colors.primaryLight};
    max-width: ${(props) => props.maxWidth || 'auto'};
  }

  small {
    margin-top: 0.3rem;
    height: 18px;
  }
`
