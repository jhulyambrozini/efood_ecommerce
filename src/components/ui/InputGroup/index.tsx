import { DetailedHTMLProps, InputHTMLAttributes } from 'react'
import { InputGroupContainer } from './styles'
import InputMask from 'react-input-mask'

export type InputGroupProps = DetailedHTMLProps<
  InputHTMLAttributes<HTMLInputElement>,
  HTMLInputElement
> & {
  label: string
  id: string
  maxWidth?: string
  mask?: string
}

const InputGroup = ({
  label,
  id,
  children,
  maxWidth,
  mask,
  ...props
}: InputGroupProps) => {
  return (
    <InputGroupContainer maxWidth={maxWidth}>
      <label htmlFor={id}>{label}</label>
      {mask ? (
        <InputMask name={id} id={id} type="text" mask={mask} {...props} />
      ) : (
        <input name={id} id={id} type="text" {...props} />
      )}

      {children}
    </InputGroupContainer>
  )
}

export default InputGroup
