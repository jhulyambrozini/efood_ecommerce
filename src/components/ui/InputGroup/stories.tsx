import type { Meta } from '@storybook/react'

import InputGroup, { InputGroupProps } from '.'

const meta = {
  title: 'ui/InputGroup',
  component: InputGroup,
  parameters: {
    layout: 'centered',
    backgrounds: {
      default: 'dark'
    }
  },
  tags: ['autodocs']
} satisfies Meta<typeof InputGroup>

// eslint-disable-next-line react-refresh/only-export-components
export default meta

export const Default = (args: InputGroupProps) => {
  return (
    <InputGroup {...args} label="Endereço" id="adress">
      {args.children}
    </InputGroup>
  )
}

export const withMask = (args: InputGroupProps) => {
  return (
    <InputGroup
      {...args}
      mask="9999 9999 9999 99999"
      label="Número no cartão"
      id="cardNumber"
    >
      {args.children}
    </InputGroup>
  )
}
