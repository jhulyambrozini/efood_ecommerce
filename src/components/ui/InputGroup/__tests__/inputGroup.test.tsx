import '@testing-library/jest-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import InputGroup from '..'

describe('<InputGroup />', () => {
  it('should render correctly', () => {
    render(<InputGroup label="name" id="name" />)
    expect(screen.getByRole('input-group')).toBeInTheDocument()
  })

  it('should be a input with a mask', () => {
    render(<InputGroup label="name" id="name" mask="9-9" />)

    const input = screen.getByRole('textbox')
    fireEvent.change(input, {
      target: {
        value: '9-9'
      }
    })
    expect(input).toHaveValue('9-9')
  })

  it('should be a input without a mask', () => {
    render(<InputGroup label="name" id="name" />)

    const input = screen.getByRole('textbox')
    fireEvent.change(input, {
      target: {
        value: 'testando'
      }
    })
    expect(input).toHaveValue('testando')
  })

  it('should render a chidren message', () => {
    render(
      <InputGroup label="name" id="name">
        mensagem de erro
      </InputGroup>
    )
    expect(screen.getByText('mensagem de erro'))
  })

  it('should be render with a maxWidth=93px', () => {
    render(<InputGroup label="name" id="name" maxWidth="93px" />)

    const input = screen.getByRole('textbox')
    expect(input).toHaveStyle('max-width: 93px')
  })

  it('should match snapshot', () => {
    const { container } = render(
      <InputGroup label="name" id="name" maxWidth="96px">
        mensagem de erro
      </InputGroup>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
