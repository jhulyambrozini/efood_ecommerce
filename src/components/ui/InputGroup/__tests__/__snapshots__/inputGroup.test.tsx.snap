// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputGroup /> should match snapshot 1`] = `
<div
  class="sc-dcJsrY jnVQvq"
  role="input-group"
>
  <label
    for="name"
  >
    name
  </label>
  <input
    id="name"
    name="name"
    type="text"
  />
  mensagem de erro
</div>
`;
